labgrid/driver/shelldriver.py

```diff
@@ -536,3 +538,55 @@ def run_script_file(self, scriptfile: str, *args, timeout: int = 60):
             IOError: if the provided localfile could not be found
         """
         return self._run_script_file(scriptfile, *args, timeout=timeout)
+
+    @Driver.check_active
+    def get_default_interface_device_name(self, version=4):
+        """ Retrieve the default route's interface device name.
+
+        Args:
+            version (int): IP version
+
+        Returns:
+            Name of the device of the default route
+
+        Raises:
+            ExecutionError: if no or multiple routes are set up
+        """
+        assert version in (4, 6)
+
+        regex = r"""default\s+via # leading strings
+                \s+\S+ # IP address
+                \s+dev\s+(\w+) # interface"""
+
+        default_route = self._run_check(f"ip -{version} route list default")
+        matches = re.findall(regex, "\n".join(default_route), re.X)
+
+        if not matches:
+            raise ExecutionError(f"No IPv{version} default route found")
+        if len(matches) > 1:
+            raise ExecutionError(f"Multiple IPv{version} default routes found")
+
+        return matches[0]
+
+    @Driver.check_active
+    def get_ip_interfaces(self, device=None):
+        """ Retrieves IP interfaces for given interface name.
```

Comment:
I find the function name confusing. It returns the IP *addresses* of a single network *interface*. @Bastian-Krause probably chose `get_ip_interfaces` because the ipaddress modules has a class name `ipaddress.IPv4/6Interface`. This is a subclass of the corresponding `IPv4/6Address`, so it'd suggest to call it `get_ip_addresses`.

Reply:
Right, so should we still return IPv4Interface/IPv6Interface objects here or rather IPv4Address/IPv6Address?

Reply:
Yes, the just contain additional information (the prefix length).